haskell-src/Concordium/Types/Parameters.hs

```diff
@@ -407,59 +1115,47 @@ data TimeParameters (cpv :: ChainParametersVersion) where
           -- |Mint rate per payday (as a proportion of the extant supply).
           _tpMintPerPayday :: !MintRate
         } ->
-        TimeParameters 'ChainParametersV1
-
--- |Lens for '_tpRewardPeriodLength'
-{-# INLINE tpRewardPeriodLength #-}
-tpRewardPeriodLength :: Lens' (TimeParameters 'ChainParametersV1) RewardPeriodLength
-tpRewardPeriodLength =
-    lens _tpRewardPeriodLength (\tp x -> tp{_tpRewardPeriodLength = x})
+        TimeParameters
+    deriving (Eq, Show)
 
--- |Lens for '_tpMintPerPayday'
-{-# INLINE tpMintPerPayday #-}
-tpMintPerPayday :: Lens' (TimeParameters 'ChainParametersV1) MintRate
-tpMintPerPayday =
-    lens _tpMintPerPayday (\tp x -> tp{_tpMintPerPayday = x})
+makeClassy ''TimeParameters
```

Comment:
Explain in detail waht functions and classes this generates. The documentation of makeClassy is poor.